src/bundle/Resources/public/js/scripts/core/dropdown.js

```diff
@@ -1,94 +1,86 @@
 (function(global, doc, eZ) {
-    const CLASS_DROPDOWN_OVERFLOW = 'ibexa-dropdown--overflow';
-    const CLASS_DROPDOWN_EXPANDED = 'ibexa-dropdown--is-expanded';
-    const CLASS_ITEMS_POSITION_TOP = 'ibexa-dropdown__items--position-top';
-    const CLASS_REMOVE_SELECTION = 'ibexa-dropdown__remove-selection';
-    const CLASS_ITEM_SELECTED_IN_LIST = 'ibexa-dropdown__item--selected';
-    const CLASS_ITEM_PREFERRED_CHOICE = 'ibexa-dropdown__item--preferred-choice';
-    const CLASS_ITEM_HIDDEN = 'ibexa-dropdown__item--hidden';
-    const SELECTOR_ITEM = '.ibexa-dropdown__item';
-    const SELECTOR_DROPDOWN = '.ibexa-dropdown';
-    const SELECTOR_SOURCE = '.ibexa-dropdown__source .ibexa-input';
-    const SELECTOR_ITEMS_CONTAINER = '.ibexa-dropdown__items';
-    const SELECTOR_ITEMS_LIST_CONTAINER = '.ibexa-dropdown__items-list';
-    const SELECTOR_ITEMS_FILTER = '.ibexa-dropdown__items-filter';
-    const SELECTOR_SELECTED_ITEM_IN_LABEL = '.ibexa-dropdown__selected-item';
-    const SELECTOR_SELECTED_ITEM_IN_LIST = '.ibexa-dropdown__item--selected';
-    const SELECTOR_SEPARATOR = '.ibexa-dropdown__separator';
-    const SELECTOR_SELECTION_INFO = '.ibexa-dropdown__selection-info';
-    const SELECTOR_OVERFLOW_ITEM_INFO = '.ibexa-dropdown__selected-overflow-number';
-    const SELECTOR_PLACEHOLDER = '.ibexa-dropdown__selected-placeholder';
     const EVENT_VALUE_CHANGED = 'change';
-    const ITEMS_LIST_MAX_HEIGHT = 300;
+    const ITEMS_LIST_MIN_HEIGHT = 300;
     const RESTRICTED_AREA_ITEMS_CONTAINER = 190;
     const MINIMUM_LETTERS_TO_FILTER = 3;
 
     class Dropdown {
         constructor(config = {}) {
-            const container = config.container ?? doc.querySelector(SELECTOR_DROPDOWN);
-            const sourceInput = config.sourceInput ?? container.querySelector(SELECTOR_SOURCE);
-
-            this.container = container;
-            this.sourceInput = sourceInput;
-            this.itemsContainer = config.itemsContainer ?? container.querySelector(SELECTOR_ITEMS_CONTAINER);
-            this.itemsListContainer = config.itemsListContainer ?? container.querySelector(SELECTOR_ITEMS_LIST_CONTAINER);
-            this.itemsFilterInput = config.itemsFilterInput ?? container.querySelector(SELECTOR_ITEMS_FILTER);
-            this.hasDefaultSelection = config.hasDefaultSelection || false;
-            this.selectedItemTemplate =
-                config.selectedItemTemplate ||
-                `<li
-                class="ibexa-dropdown__selected-item"
-                data-value="{{value}}">
-                    {{label}}<span class="${CLASS_REMOVE_SELECTION}"></span>
-            </li>`;
-            this.canSelectOnlyOne = !sourceInput?.multiple;
+            this.container = config.container.classList.contains('ibexa-dropdown') ? config.container : config.container.querySelector('.ibexa-dropdown');
+
+            if (!this.container) {
+                this.error = true;
```

Comment:
I don't think it's needed, given we throw the exception below.
It's only used in one place, in the `init` function, to check if the error occurred, but `init` should not be called after the throw in the first place. If someone catches the error and still calls `init` it's 100% his fault.